src/main/java/de/tum/cit/aet/artemis/core/web/admin/AdminOldDataCleanupResource.java

```diff
@@ -0,0 +1,111 @@
+package de.tum.cit.aet.artemis.core.web.admin;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.dto.CleanupServiceExecutionRecordDTO;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAdmin;
+import de.tum.cit.aet.artemis.core.service.cleanup.DataCleanupService;
+
+/**
+ * REST controller for managing old data cleanup operations in Artemis.
+ * Provides endpoints for administrators to clean up old or orphaned data in the database.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/admin/")
```

Comment:
Shouldn't we add another level to the endpoint?
```suggestion
@RequestMapping("api/admin/cleanup")
```